### DISCUSSION TITLE
Enable xmlreader, xmlwriter extensions

### DISCUSSION THREAD

**Discussion by bnzo** [NativePHP/laravel#379] [Ideas]
Using [Laravel Excel](https://laravel-excel.com), generating an xlsx export file throws a `Class "XMLWriter" not found` Error.
Enabling xml extensions will allow xlsx files import/export using this package.

Reading the Laravel Excel [requirements](https://docs.laravel-excel.com/3.1/getting-started/installation.html#requirements), `iconv` extension might also be needed for some use cases.

Regarding xml manipulation in general, `xsl` extension might also be useful.

**Comment by simonhamp** [comment]
Duplicate of #309

**Comment by bnzo** [comment]
@simonhamp Hey sorry for the duplicate, I will build the binaries as you recommended here #309

**Reply by simonhamp** [reply]
No worries. Hope you can get things built the way you need!

**Comment by bnzo** [comment]
@simonhamp, I built a binary (at least for my OS) and it works well.

For future reference, here are steps I followed to build a PHP binary for macOS arm64:

```sh
# Clone static-php-cli repository
git clone https://github.com/crazywhalecc/static-php-cli.git

cd static-php-cli
```

```sh
# Download PHP sources
bin/spc download --with-php=8.3 --for-extensions "bcmath,ctype,curl,dom,fileinfo,filter,gd,mbstring,opcache,openssl,pdo,pdo_sqlite,phar,session,simplexml,sockets,sqlite3,tokenizer,xml,xmlreader,xmlwriter,zip,zlib"
```

In my specific case, I had missing commands, `cmake` and `autoconf`, I installed them using `homebrew`

```sh
# https://formulae.brew.sh/formula/cmake
brew install cmake

# https://formulae.brew.sh/formula/autoconf
brew install autoconf
```

```sh
# Build
bin/spc build --build-cli --build-embed "bcmath,ctype,curl,dom,fileinfo,filter,gd,mbstring,opcache,openssl,pdo,pdo_sqlite,phar,session,simplexml,sockets,sqlite3,tokenizer,xml,xmlreader,xmlwriter,zip,zlib"
```

```sh
# Delete all downloaded sources
bin/spc download --clean
```

Then following NativePHP documentation section [Building apps with custom binaries](https://nativephp.com/docs/1/digging-deeper/php-binaries#building-apps-with-custom-binaries) to respect directories structure and that is pretty much it.
